core/Db.php

```diff
@@ -285,12 +286,55 @@ public static function query($sql, $parameters = array())
         }
     }
 
+    /**
+     * Executes a callback with potential recovery from a "MySQL server has gone away" error.
+     *
+     * If the callback throws a "MySQL server has gone away" exception
+     * it will be called again after a single reconnection attempt.
+     *
+     * @param callable $callback
+     *
+     * @return mixed
+     *
+     * @throws Exception
+     *
+     * @internal
+     */
+    public static function executeWithDatabaseWriterReconnectionAttempt(callable $callback)
+    {
+        try {
+            return $callback();
+        } catch (Exception $ex) {
+            // only attempt reconnection in a reader/writer configuration
+            if (!self::hasReaderConfigured()) {
+                throw $ex;
+            }
+
+            // only attempt reconnection if we encounter a "MySQL server has gone away" error
+            if (
+                !self::get()->isErrNo($ex, Updater\Migration\Db::ERROR_CODE_MYSQL_SERVER_HAS_GONE_AWAY)
+                && false === stripos($ex->getMessage(), 'MySQL server has gone away')
```

Comment:
Do MariaDb and Aurora throw the same message/same wording of the message?

Reply:
The message should be triggered by the client library and not depend on the actual server connected to.

However I changed the actual wording tested to only `server has gone away`, to keep it less MySQL specific. Sometime in the future the underlying client library may change to MariaDB and they apparently [change](https://us02web.zoom.us/j/6827118593) the wording :thinking: